app/controllers/search_controller.rb

```diff
@@ -25,9 +25,10 @@ def index
 
     # filter out devel projects on user setting
     unless (@search_unsupported || @search_project)
-      @packages = @packages.select { |p| (@distributions.map { |d| d[:project] }.include? p.project) ||
+      @packages = @packages.select do |p|
+        @distributions.map { |d| d[:project] }.include?(p.project) ||
           @distributions.map { |d| "#{d[:project]}:Update" }.include?(p.project) || @distributions.map { |d| "#{d[:project]}:NonFree" }.include?(p.project)
```

Comment:
:see_no_evil: :see_no_evil: :see_no_evil:

What about using [`flat_map`](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-flat_map) to change lines 29 and 30 by:

``` ruby
@distributions.flat_map { |d| [d[:project], "#{d[:project]}:Update", "#{d[:project]}:NonFree"] }.include? p.project
```

this should be not only more readable but also more efficient :wink: